File: src/main/java/burrows/apps/gif/example/di/module/RiffsyModule.java
package burrows.apps.gif.example.di.module;


import burrows.apps.gif.example.di.scope.PerActivity;
import burrows.apps.gif.example.rest.service.RiffsyRepository;
import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

/**
 * @author <a href="mailto:jaredsburrows@gmail.com">Jared Burrows</a>
 */
@Module
public class RiffsyModule {
  @Provides @PerActivity protected RiffsyRepository provideRiffsyService(Retrofit.Builder retrofit) {
    return new RiffsyRepository(retrofit);
  }
}

File: src/main/java/com/burrowsapps/example/gif/di/module/RiffsyModule.kt
package com.burrowsapps.example.gif.di.module

import com.burrowsapps.example.gif.BuildConfig
import com.burrowsapps.example.gif.data.RiffsyApiClient
import dagger.Module
import dagger.Provides
import retrofit2.Retrofit

@Module
class RiffsyModule(private var baseUrl: String = BuildConfig.BASE_URL) {
    @Provides fun providesRiffsyApi(retrofit: Retrofit.Builder): RiffsyApiClient = retrofit
        .baseUrl(baseUrl)
        .build()
        .create(RiffsyApiClient::class.java)
}
